src/java/org/apache/cassandra/net/AsyncStreamingInputPlus.java

```diff
@@ -115,7 +115,8 @@ protected void reBuffer() throws ClosedChannelException, InputTimeoutException
         if (queue.isEmpty())
             channel.read();
 
-        currentBuf.release();
+        if (currentBuf != null)
```

Comment:
this was found during code inspection, if you call rebuffed and get a timeout, then call rebuffed again this is null.

Reply:
If that happens, is there any case where we don't immediately call `close()` (which does check to see if the `currentBuf` is `null`)? I'm just a little worried that we actually shouldn't ever try to `reBuffer()` after that happens, and not throwing a runtime exception if that happens is actually the opposite of what we want...

Reply:
> is there any case where we don't immediately call close()

don't know

> and not throwing a runtime exception if that happens is actually the opposite of what we want...

fail, keeping it NPE does help